src/Microsoft.ServiceHub.Framework/Container/GlobalBrokeredServiceContainer.cs

```diff
@@ -286,6 +286,13 @@ internal async ValueTask<ServiceRpcDescriptor> ApplyDescriptorSettingsInternalAs
 			.WithJoinableTaskFactory(this.joinableTaskFactory)
 			.WithTraceSource(traceSource);
 
+		if (descriptor is ServiceJsonRpcDescriptor { AdditionalServiceInterfaces.Length: 0 } jsonRpcDescriptor &&
```

Comment:
Does this mean that if a component presents a descriptor with no additional interfaces but service registration has some, the returned proxy will have those interfaces? But if there are entries in the descriptor provided, the registration entries will not be used?

I am wondering if we should combine the lists here instead?

Reply:
You are correct in how you read this.

Merging the lists was my first inclination. And maybe it's the right one. But I got to thinking that if some client provides a descriptor with an explicit list of interfaces, that's very advanced and I might well assume they know what they're doing. They might want to _remove_ interfaces from the registered list.

The remaining question for me then was whether there should be a way to avoid the registered list while not providing any additional interfaces of your own. Right now, the client can override the registered list only by providing a non-empty array of interfaces. I'm thinking this is less likely, and if they really needed it, they could declare an empty, internal interface and put it in the array to effectively remove any noticeable interface from the proxy.

Thoughts?

Reply:
Could we keep the list as nullable and treat "null" as a hint to use the interfaces from service registration and then empty array would still be considered as an override?

The one scenario I can think of where a client may really want to control these additional interfaces is when they try to avoid loading additional types (thus modules) but I am not really sure how likely it would be that interfaces span across multiple assemblies.

Maybe we can add a bool property in future if this becomes a requirement?

Reply:
Nullable was another option I considered. Given you suggested it as well, and my desire to avoid adding a bool property later, I'll go ahead and make that change.